modules/org.restlet.ext.apispark/src/org/restlet/ext/apispark/internal/conversion/swagger/v2_0/Swagger2Translator.java

```diff
@@ -581,4 +617,523 @@ public static Swagger getSwagger(Definition definition) {
+        definition.setContract(contract);
+
+        // Media types
+        if (swagger.getProduces() != null) {
+            produces.addAll(swagger.getProduces());
+        }
+        if (swagger.getConsumes() != null) {
+            consumes.addAll(swagger.getConsumes());
+        }
+
+        // General
+        definition.setVersion(info.getVersion());
+        definition.setTermsOfService(info.getTermsOfService());
+        // TODO: attribution and keywords not implemented in Swagger 2.0
+        definition.setAttribution(null);
+        definition.setKeywords(new ArrayList<String>());
+    }
+
+    private static void fillRepresentations(Swagger swagger, Contract contract) {
+        if (swagger.getDefinitions() == null) {
+            return;
+        }
+
+        for (String key : swagger.getDefinitions().keySet()) {
+            Representation representation = new Representation();
+            Model model = swagger.getDefinitions().get(key);
+            representation.setDescription(model.getDescription());
+            representation.setName(key);
+            representation.setRaw(false);
+            // TODO: example not implemented in RWADef (built from properties examples)
+            fillRwadefProperties(model, representation);
+            contract.getRepresentations().add(representation);
+        }
+    }
+
+    private static void fillRwadefProperties(Model model, Representation representation) {
+        if (model.getProperties() == null) {
+            return;
+        }
+
+        sortSwaggerPropertiesByPosition(model);
+
+        for (String key : model.getProperties().keySet()) {
+            org.restlet.ext.apispark.internal.model.Property property =
+                    new org.restlet.ext.apispark.internal.model.Property();
+            Property swaggerProperty = model.getProperties().get(key);
+
+            property.setDefaultValue(swaggerProperty.getDefault());
+            property.setDescription(swaggerProperty.getDescription());
+            // TODO: enumeration not implemented in Swagger 2.0
+            property.setExample(swaggerProperty.getExample());
+            property.setMinOccurs(swaggerProperty.getRequired() ? 1 : 0);
+            property.setMaxOccurs(swaggerProperty instanceof ArrayProperty ? -1 : 1);
+            property.setName(key);
+            // TODO: sub-properties not implemented in Swagger 2.0
+            // TODO: uniqueItems not implemented in Swagger 2.0
+            property.setUniqueItems(false);
+
+            if (swaggerProperty instanceof AbstractNumericProperty) {
+                AbstractNumericProperty numericProperty = (AbstractNumericProperty) swaggerProperty;
+                formatPropertyMinMax(numericProperty, property);
+            } else if (swaggerProperty instanceof ArrayProperty) {
+                ArrayProperty arrayProperty = (ArrayProperty) swaggerProperty;
+                property.setExample(arrayProperty.getItems().getExample());
+            }
```

Comment:
The switch is only to fill in properties that are specific to a certain type. 
So well, yes continue.